tests/tensor/nnet/test_basic.py

```diff
@@ -1028,39 +1038,61 @@ def test_grad(self):
         # test that function contains softmax and softmaxgrad
         w = matrix()
 
-        g = aesara.function([c, w], grad((p_y * w).sum(), c))
+        g = aesara.function([c, w], grad((p_y * w).sum(), c), mode=self.mode)
 
         g_ops = [n.op for n in g.maker.fgraph.toposort()]
 
-        assert len(g_ops) == 2
-        assert softmax_legacy in g_ops
-        assert softmax_grad_legacy in g_ops
+        assert len(g_ops) == 2, g_ops
+        assert isinstance(g_ops[0], Softmax)
+        assert isinstance(g_ops[1], SoftmaxGrad)
 
         g(self.rng.random((3, 4)), self.rng.uniform(0.5, 1, (3, 4)))
 
-    @pytest.mark.skip(reason="Optimization not enabled for the moment")
     def test_transpose_basic(self):
         # this should be a transposed softmax
         c = matrix()
         p_y = exp(c) / exp(c).sum(axis=0)
 
         # test that function contains softmax and no div.
-        aesara.function([c], p_y)
+        f = aesara.function([c], p_y, mode=self.mode)
+        f_ops = [n.op for n in f.maker.fgraph.toposort()]
+        assert len(f_ops) == 1
+        assert isinstance(f_ops[0], Softmax)
+
+    @pytest.mark.skip(reason="Optimization not enabled for the moment")
```

Comment:
I couldn't find any existing rewrite that is supposed to cover this situation: `grad(softmax_graph) -> SoftmaxGrad` where `softmax_graph` is the symbolic softmax expression not using the Softmax Op directly.

Reply:
This was brought up in the second point of this old Theano issue: https://github.com/Theano/Theano/issues/4452